flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -86,37 +89,90 @@
 
 	private final ResultPartitionManager resultPartitionManager;
 
+	private final Map<IntermediateDataSetID, SingleInputGate> inputGatesById;
```

Comment:
Maybe it is more clearly to define `ConcurrentHashMap` instead of `Map` here.

Reply:
Let's keep `Map` because `NetworkEnviroment` does not depend on `ConcurrentHashMap` methods  at the moment.